app/packs/src/apps/mydb/elements/details/researchPlans/analysesTab/ResearchPlanDetailsContainers.js

```diff
@@ -14,101 +14,8 @@ import SpectraActions from 'src/stores/alt/actions/SpectraActions';
 import LoadingActions from 'src/stores/alt/actions/LoadingActions';
 import ViewSpectra from 'src/apps/mydb/elements/details/ViewSpectra';
 import NMRiumDisplayer from 'src/components/nmriumWrapper/NMRiumDisplayer';
-
 import TextTemplateActions from 'src/stores/alt/actions/TextTemplateActions';
-
-const SpectraEditorBtn = ({
-  element, spcInfos, hasJcamp, hasChemSpectra,
-  toggleSpectraModal, confirmRegenerate,
-  toggleNMRDisplayerModal, hasNMRium,
-}) => (
-  <span>
-  <OverlayTrigger
-    placement="bottom"
-    delayShow={500}
-    overlay={<Tooltip id="spectra">Spectra Editor {spcInfos.length > 0 ? '' : ': Reprocess'}</Tooltip>}
-  >{spcInfos.length > 0 ? (
-    <ButtonGroup className="button-right">
-      <SplitButton
-        id="spectra-editor-split-button"
-        pullRight
-        bsStyle="info"
-        bsSize="xsmall"
-        title={<i className="fa fa-area-chart" />}
-        onToggle={(open, event) => { if (event) { event.stopPropagation(); } }}
-        onClick={toggleSpectraModal}
-        disabled={!(spcInfos.length > 0) || !hasChemSpectra}
-      >
-        <MenuItem
-            id="regenerate-spectra"
-            key="regenerate-spectra"
-            onSelect={(eventKey, event) => {
-              event.stopPropagation();
-              confirmRegenerate(event);
-            }}
-            disabled={!hasJcamp || !element.can_update}
-          >
-            <i className="fa fa-refresh" /> Reprocess
-          </MenuItem>
-        </SplitButton>
-      </ButtonGroup>
-      ) : (
-        <Button
-          bsStyle="warning"
-          bsSize="xsmall"
-          className="button-right"
-          onClick={confirmRegenerate}
-          disabled={!hasJcamp || !element.can_update || !hasChemSpectra}
-        >
-          <i className="fa fa-area-chart" /><i className="fa fa-refresh " />
-      </Button>
-    )}
-    </OverlayTrigger>
-    {
-      hasNMRium ? (
-        <OverlayTrigger
-          placement="top"
-          delayShow={500}
-          overlay={<Tooltip id="spectra_nmrium_wrapper">Process with NMRium</Tooltip>}
-        >
-          <ButtonGroup className="button-right">
-            <Button
-              id="spectra-editor-split-button"
-              pullRight
-              bsStyle="info"
-              bsSize="xsmall"
-              onToggle={(open, event) => { if (event) { event.stopPropagation(); } }}
-              onClick={toggleNMRDisplayerModal}
-              disabled={!hasJcamp}
```

Comment:
Note that the `disabled` flag of the NMRium button did not depend on `element.can_update` until now.
For `Reaction` and `Sample` the availability of the NMRium button *does* depend on `element.can_update`. 
Is there a reason for treating `ResearchPlan`s as a special case?

Reply:
@JanCBrammer The `NMRium` button is still disabled on `ResearchPlan`. 
For the `element.can_update`, only `SampleEntity` and `ReactionEntity` have the value `can_update` while `ResearchPlanEntity` doesn't have it. You can check the `sample_entity.rb`, `reaction_entity.rb`  and  `research_plan_entity.rb` to see that.

Reply:
> The `NMRium` button is still disabled on `ResearchPlan`

Ok, so the `NMRium` button is *supposed* to be disabled for research plans?

Reply:
Yes, in the case that there is not `NMRium` service. So you can ignore the `can_update` value for `ResearchPlan` because it's always `true`